java/runtime/src/main/java/io/ray/runtime/config/RayConfig.java

```diff
@@ -111,8 +111,9 @@ public RayConfig(Config config) {
     if (config.hasPath("ray.node-ip")) {
       nodeIp = config.getString("ray.node-ip");
     } else {
-      nodeIp = NetworkUtil.getIpAddress(null);
+      nodeIp = NetworkUtil.localhostIp();
```

Comment:
How is this nodeIp used? On Linux, the node ip is by default public ip instead of localhost in cpp and python.

Reply:
This is used for 2 place right now:
First one is https://github.com/ray-project/ray/blob/master/java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java#L93 . In driver process, it will used for getting current node info from GCS by the ip.
The 2nd place is https://github.com/ray-project/ray/blob/master/java/runtime/src/main/java/io/ray/runtime/RayNativeRuntime.java#L134 .  We used it as a field of `CoreWorker.Options` for constructing a coreworkerprocess instance .

It seems we need pass a public ip on linux.

Reply:
I updated the code. Feel free to take more looks. :)